src/components/Article.jsx

```diff
@@ -32,19 +53,51 @@ class Article extends React.Component {
     this.state = {
       article: {},
       bookmarked: false,
+      token: '',
+      recommendations: [],
+    };
+  }
+
+  /**
+   * @static
+   * @param {*} nextProps - The new props received by the component
+   * @returns {Object} - The article object to display in the page
+   * @memberof Article
+   */
+  static getDerivedStateFromProps(nextProps) {
+    let bookmarked;
+    const {
+      token, recommendations, bookmarkedList: bookmarkedArticles, id,
+    } = nextProps;
+    const article = { ...nextProps };
+    delete article.match;
+    delete article.location;
+    delete article.history;
+    delete article.getArticle;
+    delete article.staticContext;
+    delete article.token;
+    bookmarkedArticles.forEach((post) => {
+      if (post.articleId === id) bookmarked = true;
```

Comment:
What do you think about this?

```suggestion
bookmarked = post.articleId === id
```
Also we may consider using Array.findIndex(), this way the loop stops immediately the `if` statement returns `true`